zookeeper-server/src/main/java/org/apache/zookeeper/server/auth/X509AuthenticationConfig.java

```diff
@@ -0,0 +1,324 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.zookeeper.server.auth;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Set;
+import java.util.stream.Collectors;
+import com.google.common.annotations.VisibleForTesting;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This class contains config properties for x509-based authentication
+ */
+public class X509AuthenticationConfig {
+  private static final Logger LOG = LoggerFactory.getLogger(X509AuthenticationConfig.class);
+  private static X509AuthenticationConfig instance = null;
+  private static final String LOG_MSG_PREFIX = X509AuthenticationConfig.class.getName() + "::";
+
+  private X509AuthenticationConfig() {
+  }
+
+  public static X509AuthenticationConfig getInstance() {
+    if (instance == null) {
+      synchronized (X509AuthenticationConfig.class) {
+        if (instance == null) {
+          instance = new X509AuthenticationConfig();
+        }
+      }
+    }
+    return instance;
+  }
+
+  /**
+   * The following System Property keys are used to extract clientId from the client cert.
+   */
+  public static final String SSL_X509_CONFIG_PREFIX = "zookeeper.ssl.x509.";
```

Comment:
Does this break backward-compatibility (from previous versions)? If not, I think this is okay.

Reply:
This is different from previous versions, but this change is inevitable since these configs starts with `zookeeper.X509AuthenticationProvider.` in previous versions, but right now we are moving them out of X509AuthenticationProvider.

Reply:
Does this sound okay to you? @narendly

Reply:
@mgao0 normally, we should ensure that older configs continue to work, but in this case since the build versions are so close together that I'm okay with this. But for the future, let's make sure we add logic that picks up old config keys as well.